osu.Game/Rulesets/UI/GameplaySampleTriggerSource.cs

```diff
@@ -56,12 +56,12 @@ public virtual void Play()
             PlaySamples(samples);
         }
 
-        protected void PlaySamples(ISampleInfo[] samples)
+        protected void PlaySamples(ISampleInfo[] samples) => Schedule(() =>
```

Comment:
> the scheduled playback will still be processed in the same frame

I don't... think so? In empirical testing `PlaySamples()` is called via `Column.OnPressed()`, on the update thread. At that point the input has already made an update frame window, so this schedule *will* incur a one-frame delay (and it cannot be processed inline, because that would ruin the fix).

Reply:
As far as I have been led to believe, input is processed before `UpdateSubTree`. If that's not the case, at least one other of my recent changes is incorrect.

Reply:
Actually on second thought I sort of take it back? Scheduling to a child is safe as long as the task is enqueued onto the child's scheduler. But in this case it isn't, because a naked `Schedule()` call implicitly enqueues onto *this drawable's* scheduler. So I think this is still a delay, unless my rationale for why this is OK is entirely wrong?

Reply:
The child in this case would be the `GameplaySampleTriggerSource`, since the point of scheduling is all happening from `InputManager.Update`.

In other words, imagine the following happening at an update frame, after a user pressed a key:
 - `InputManager.Update()`
    - Iterates on pending input, applies pressed key, and propagate `OnKeyDown` event.
    - `KeyBindingContainer` receives it and propagates `OnPressed` event.
    - `Column` receives it and calls `GameplaySampleTriggerSource.Play`.
    - `GameplaySampleTriggerSource.Play` schedules the sample playback.
 - `GameplaySampleTriggerSource.Update()`
    - Processes the scheduler and invokes the sample playback delegate

Therefore the delay is at least not for a full frame.